BiometricLoginKotlin/app/src/main/java/com/example/biometricloginsample/EnableBiometricLoginActivity.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+package com.example.biometricloginsample
+
+import android.content.Context
+import android.os.Bundle
+import android.util.Log
+import android.view.inputmethod.EditorInfo
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.biometric.BiometricManager
+import androidx.biometric.BiometricPrompt
+import androidx.lifecycle.Observer
+import com.example.biometricloginsample.databinding.ActivityEnableBiometricLoginBinding
+
+class EnableBiometricLoginActivity : AppCompatActivity() {
+    private val TAG = "EnableBiometricLogin"
+    private lateinit var cryptographyManager: CryptographyManager
+    private val loginViewModel by viewModels<LoginWithPasswordViewModel>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val binding = ActivityEnableBiometricLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        binding.cancel.setOnClickListener { finish() }
+
+        loginViewModel.loginWithPasswordFormState.observe(this, Observer {
```

Comment:
From here onward the content of `onCreate` could be spread out in separate functions, making `onCreate` cleaner to read.

Reply:
Yeah, I mentally played around with it. Right now `onCreate` is one of four meaningful methods. I did a quick cost-benefit analysis: if I pulled out lines 58-76 then I'd have to do the same for lines 51-56, at which point it started looking like overkill. I mentally played with a few other rearrangements and couldn't really justify them. As a general rule I could split it, but I wasn't convinced the reader would have difficulty reading through this onCreate since it's mostly view binging that don't really do much else.

Reply:
`setUpInterface` is a simple way to put this, which doesn't require parsing the entirety of `onCreate`.
It makes a difference, especially for folks that are just getting started. Please extract it in with a meaningful function name.

Reply:
Your original idea was to "spread out in separate functions", which would create more complexity than value. This new idea of taking lines 41-end out of `onCreate` and just lump them into another function still doesn't really add value. There is truly nothing else happening in `onCreate` to necessitate this split.

Reply:
Ack